sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AnalyzeSentimentAsyncClient.java

```diff
@@ -175,24 +195,135 @@ private AnalyzeSentimentResult convertToAnalyzeSentimentResult(DocumentSentiment
      * {@link AnalyzeSentimentResultCollection} from a {@link SimpleResponse} of {@link SentimentResponse}.
      *
      * @param documents A list of documents to be analyzed.
-     * @param options The {@link TextAnalyticsRequestOptions} request options.
+     * @param options The additional configurable {@link AnalyzeSentimentOptions options} that may be passed when
+     * analyzing sentiments.
      * @param context Additional context that is passed through the Http pipeline during the service call.
      *
      * @return A mono {@link Response} contains {@link AnalyzeSentimentResultCollection}.
      */
     private Mono<Response<AnalyzeSentimentResultCollection>> getAnalyzedSentimentResponse(
-        Iterable<TextDocumentInput> documents, TextAnalyticsRequestOptions options, Context context) {
-        // TODO: add opinion mining in the following PR
+        Iterable<TextDocumentInput> documents, AnalyzeSentimentOptions options, Context context) {
         return service.sentimentWithResponseAsync(
             new MultiLanguageBatchInput().setDocuments(toMultiLanguageInput(documents)),
             options == null ? null : options.getModelVersion(),
             options == null ? null : options.isIncludeStatistics(),
-            null,
+            options == null ? null : options.isIncludeOpinionMining(),
             context.addData(AZ_TRACING_NAMESPACE_KEY, COGNITIVE_TRACING_NAMESPACE_VALUE))
             .doOnSubscribe(ignoredValue -> logger.info("A batch of documents - {}", documents.toString()))
             .doOnSuccess(response -> logger.info("Analyzed sentiment for a batch of documents - {}", response))
             .doOnError(error -> logger.warning("Failed to analyze sentiment - {}", error))
             .map(this::toAnalyzeSentimentResultCollectionResponse)
             .onErrorMap(throwable -> mapToHttpResponseExceptionIfExist(throwable));
     }
+
+    /*
+     * Transform SentenceSentiment's opinion mining to output that user can use.
+     */
+    private IterableStream<MinedOpinion> toMinedOpinionList(
+        com.azure.ai.textanalytics.implementation.models.SentenceSentiment sentenceSentiment,
+        List<DocumentSentiment> documentSentimentList) {
+        // If include opinion mining indicator is false, the service return null for the aspect list.
+        final List<SentenceAspect> sentenceAspects = sentenceSentiment.getAspects();
```

Comment:
What happens if the service accidentally sends `getAspects()` as empty even when the boolean indicator for opinion mining is set to true. Seems like we wouldn't catch that. Should check for if the boolean is set not if this is empty. Service can send back error data.

Reply:
if the getAspects() is emtpy, we return an empty list of MinedOpinon.
```
final List<MinedOpinion> minedOpinions = new ArrayList<>();
return new IterableStream<>(minedOpinions);
```

Reply:
empty list could mean there is no mined opinion but it should not be an exception that need to be catch.